Scripts/DowloadPortalItemData.swift

```diff
@@ -241,39 +231,56 @@ let previousDownloadedItems: DownloadedItems = {
-    // Checks if a directory exists or not, to see if an item is already downloaded.
-    if FileManager.default.fileExists(atPath: destinationURL.path) {
-        print("info: Item \(portalItem.identifier) has already been downloaded.")
-    } else {
+await withTaskGroup(of: Void.self) { group in
+    for portalItem in portalItems {
+        // Checks to see if an item is already downloaded.
+        guard downloadedItems[portalItem.identifier] == nil else {
+            print("note: Item already downloaded: \(portalItem.identifier)")
+            continue
+        }
+        
+        let destinationURL = downloadDirectoryURL.appendingPathComponent(portalItem.identifier, isDirectory: true)
+        
+        // Deletes the directory if it already exists.
+        // This happens when the item is not in the plist and needs to be redownloaded.
+        if FileManager.default.fileExists(atPath: destinationURL.path) {
+            do {
+                print("note: Deleting directory: \(portalItem.identifier)")
+                try FileManager.default.removeItem(at: destinationURL)
+            } catch {
+                print("error: Error deleting downloaded directory: \(error.localizedDescription)")
+                exit(1)
+            }
+        }
+        
         do {
             // Creates an enclosing directory with portal item ID as its name.
             try FileManager.default.createDirectory(at: destinationURL, withIntermediateDirectories: false)
         } catch {
-            print("error: Error creating download directory: \(error.localizedDescription).")
+            print("error: Error creating download directory: \(error.localizedDescription)")
             exit(1)
         }
-        print("info: Downloading item \(portalItem.identifier)")
+        
+        print("note: Downloading item: \(portalItem.identifier)")
         fflush(stdout)
-        dispatchGroup.enter()
-        downloadFile(at: portalItem.dataURL, to: destinationURL) { result in
-            switch result {
-            case .success(let url):
-                downloadedItems[portalItem.identifier] = url.lastPathComponent
-                dispatchGroup.leave()
-            case .failure(let error):
+        
+        group.addTask {
+            do {
+                let downloadName = try await downloadFile(from: portalItem.dataURL, to: destinationURL)
+                print("note: Downloaded item: \(portalItem.identifier)")
+                fflush(stdout)
+                
+                await MainActor.run {
+                    downloadedItems[portalItem.identifier] = downloadName
+                }
+            } catch {
                 print("error: Error downloading item \(portalItem.identifier): \(error.localizedDescription)")
                 URLSession.shared.invalidateAndCancel()
                 exit(1)
             }
         }
     }
 }
-dispatchGroup.wait()
```

Comment:
[func waitForAll() async](https://developer.apple.com/documentation/swift/taskgroup/waitforall()) may be needed for the task group. Otherwise, when the last child task is added, the script moves directly to the `downloadedItems != previousDownloadedItems` equality check below, which might cause problems.